common/persistence/executionManager.go

```diff
@@ -112,6 +112,13 @@ func (m *executionManagerImpl) GetWorkflowExecution(
 	newResponse.State.VersionHistories = versionHistories
 	newResponse.MutableStateStats = m.statsComputer.computeMutableStateStats(response)
 
+	if len(newResponse.State.Checksum.Value) == 0 {
+		newResponse.State.Checksum, err = m.serializer.DeserializeChecksum(response.State.ChecksumData)
+		if err != nil {
+			return nil, err
```

Comment:
does the err indicate it's checksum deserialization that failed or will it be just "invalid character etc." kind of error? If so maybe wrap the err with more context

Reply:
The serializer provides the context.